Yank/restraints.py

```diff
@@ -50,41 +140,58 @@ class ReceptorLigandRestraint(object):
 
     NOTES
 
-    To create a subclass that uses a different restraint energy function, follow these steps:
-
-    * Redefine class variable 'energy_function' with the energy function of choice.
-    * Redefine class variable 'bond_parameter_names' to list the parameters in 'energy_function'
-      that must be computed for each system.
-    * Redefine the _determineBondParameters() member function to compute these parameters and
-      return them in a list in the same order as 'bond_parameter_names'.
+    Creating a subclass requires the following:
+    * Perform any necessary processing in subclass `__init__` after calling base class `__init__`
+    * Compute `standard_state_correction` in `__init__`
+    * Override getRestraintForce() to return a new `Force` instance imposing the restraint
+
+    Public class fields
+    -------------------
+    topology : openmm.app.Topology
+        Reference topology for the complex
+    state : simtk.openmm.State
+        The State object used during initialization
+    system : simtk.openmm.System
+        The System object used during initialization
+    positions : simtk.unit.Quantity of dimension (nparticles,3) with units compatible with angstroms
+        The system positions used during initialization to define the restraint
+        TODO: Should this be optional?
+    receptor_atoms : list of int
+        List of atoms declared to be part of the receptor
+    ligand_atoms : list of int
+        List of atoms declared to be part of the ligand to be restrained to the receptor
+    temperature : simtk.unit.Quantity
+        The temperature specified during initialization.
+    kT : simtk.unit.Quantity
+        Thermal specified during initialization.
+    beta : simtk.unit.Quantity
+        The inverse temperature specified during initialization.
+
+    TODO: Should these all really be public data fields?
```

Comment:
Good question. It sounds reasonable to allow the control of these parameters to avoid catches (for example, the current scheme to compute `K` automatically in the Harmonic restraints fails when there are only 2 atoms).

I would prefer having them accessible as public attributes `restraint.K`. It's just cleaner than a dictionary, and they can be substituted by properties in case we need to manipulate them on get/set. We can still have a read-only `restraint.parameters` property that builds a dictionary from the parameters in case that's handy to store these in the `nc` file.